addons/l10n_fr_invoice_addr/views/report_invoice.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <t t-set="o" t-value="o.with_context(lang=lang)"/>
+
+        <xpath expr="(//t[@t-set='address']/address)[1]" position="replace">
+            <t t-set="partner" t-value="(o.l10n_fr_is_company_french and o.partner_id.parent_id) or o.partner_id"/>
+            <address t-field="partner.self" class="m-0"
+                     t-options='{"widget": "contact", "fields": ["address", "name"], "no_marker": True}'/>
+
+            <t t-if="o.l10n_fr_is_company_french and partner.siret">
+                SIRET: <t t-esc="partner.siret"/>
+            </t>
+
+            <div class="mb-3"/>
+        </xpath>
+
+        <xpath expr="//div[@groups='sale.group_delivery_invoice_address'][@t-field='o.partner_id']" position="attributes">
```

Comment:
Since we have a dependecy on sale i'm not sure if it would not be better to inherit the report_invoice_document_inherit_sale and do the xpath in there just to be explicit that we are modifying this one and not another